pkg/util/moving_sum.go

```diff
@@ -0,0 +1,96 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package util
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/benbjohnson/clock"
+)
+
+type Bucket struct {
+	timestamp time.Time
+	sum       int64
+}
+
+type MovingSum struct {
+	buckets    []Bucket
+	timeWindow time.Duration
+	bucketSize time.Duration
+	clock      clock.Clock
+	lock       sync.Mutex
+}
+
+func NewMovingSum(timeWindow time.Duration, bucketSize time.Duration, clock clock.Clock) *MovingSum {
+	return &MovingSum{
+		buckets:    make([]Bucket, 0),
+		timeWindow: timeWindow,
+		bucketSize: bucketSize,
+		clock:      clock,
+		lock:       sync.Mutex{},
+	}
+}
+
+func (ms *MovingSum) AddBytes(byteValue int64) {
+	ms.lock.Lock()
+	defer ms.lock.Unlock()
+
+	ms.dropOldbuckets()
+	now := ms.clock.Now()
+	if len(ms.buckets) == 0 || ms.buckets != nil && now.Sub(ms.buckets[len(ms.buckets)-1].timestamp) >= ms.bucketSize {
```

Comment:
Can this if statement be simplified? 
I don't think `ms.buckets` will never be `nil`, so `len(ms.buckets) == 0 || ms.buckets != nil` will always evaluate to true. 

Can we rewrite this as: 

```go
if len(ms.buckets) == 0 || now.Sub(ms.buckets[len(ms.buckets)-1].timestamp) >= ms.bucketSize
```
(or something similar)?